Source/Core/Core/CheatSearch.cpp

```diff
@@ -0,0 +1,377 @@
+      results.emplace_back();
+      results.back().m_address = addr;
+      results.back().m_value = current_value;
+    }
+  });
+
+  return results;
+}
+
+Cheats::DataType Cheats::GetDataType(const Cheats::SearchValue& value)
+{
+  // sanity checks that our enum matches with our std::variant
+  static_assert(
+      std::is_same<
+          u8, std::remove_cv<std::remove_reference<decltype(
+                  std::get<static_cast<int>(DataType::U8)>(value.m_value))>::type>::type>::value);
+  static_assert(
+      std::is_same<
+          u16, std::remove_cv<std::remove_reference<decltype(
+                   std::get<static_cast<int>(DataType::U16)>(value.m_value))>::type>::type>::value);
+  static_assert(
+      std::is_same<
+          u32, std::remove_cv<std::remove_reference<decltype(
+                   std::get<static_cast<int>(DataType::U32)>(value.m_value))>::type>::type>::value);
+  static_assert(
+      std::is_same<
+          u64, std::remove_cv<std::remove_reference<decltype(
+                   std::get<static_cast<int>(DataType::U64)>(value.m_value))>::type>::type>::value);
+  static_assert(
+      std::is_same<
+          float,
+          std::remove_cv<std::remove_reference<decltype(
+              std::get<static_cast<int>(DataType::F32)>(value.m_value))>::type>::type>::value);
+  static_assert(
+      std::is_same<
+          double,
+          std::remove_cv<std::remove_reference<decltype(
+              std::get<static_cast<int>(DataType::F64)>(value.m_value))>::type>::type>::value);
+  static_assert(std::is_same<std::vector<u8>, std::remove_cv<std::remove_reference<decltype(
+                                                  std::get<static_cast<int>(DataType::ByteArray)>(
+                                                      value.m_value))>::type>::type>::value);
+
+  return static_cast<DataType>(value.m_value.index());
+}
+
+std::vector<u8> Cheats::GetValueAsByteVector(const Cheats::SearchValue& value)
+{
+  std::vector<u8> result;
+  DataType type = GetDataType(value);
+  switch (type)
+  {
+  case Cheats::DataType::U8:
+    result.push_back(std::get<u8>(value.m_value));
+    break;
+  case Cheats::DataType::U16:
+  {
+    u16 v = std::get<u16>(value.m_value);
+    result.push_back((v >> 8) & 0xFF);
+    result.push_back(v & 0xFF);
+    break;
+  }
+  case Cheats::DataType::U32:
+  case Cheats::DataType::F32:
+  {
+    u32 v;
+    if (type == Cheats::DataType::F32)
+      v = Common::BitCast<float, u32>(std::get<float>(value.m_value));
+    else
+      v = std::get<u32>(value.m_value);
+    result.push_back((v >> 24) & 0xFF);
+    result.push_back((v >> 16) & 0xFF);
+    result.push_back((v >> 8) & 0xFF);
+    result.push_back(v & 0xFF);
+    break;
+  }
+  case Cheats::DataType::U64:
+  case Cheats::DataType::F64:
+  {
+    u64 v;
+    if (type == Cheats::DataType::F64)
+      v = Common::BitCast<double, u64>(std::get<double>(value.m_value));
```

Comment:
This will not perform the intended behavior. You need to do `Common::BitCast<u64>(...)`